File: 7KYU/Stacked Balls - 2D/solution.clj
(ns kata)
(defn ** [n]
  (* n n))

(defn stack-height-2d [n]
  (if (zero? n)
    0
    (+ 1 (Math/sqrt (- (** (- n 1.0))
                       (** (/ (- n 1.0) 2.0)))))))

File: 7KYU/Stacked Balls - 2D/solution.go
package kata

import "math"

func StackHeight2d(layers int) float64 {
	if layers == 0 {
		return 0
	} else {
		return math.Sqrt(math.Pow(float64((layers-1)), 2)-math.Pow(((float64(layers-1))/2), 2)) + 1
	}
}
